File: src/ts/magma/app/compile/type/TemplateNode.ts
import { Node } from "../../../../magma/app/compile/node/Node";
import { Iterable } from "../../../../magma/api/collect/list/Iterable";
import { Merger } from "../../../../magma/app/compile/merge/Merger";
import { ValueMerger } from "../../../../magma/app/compile/merge/ValueMerger";
export class TemplateNode implements Node {
	base: string;
	args: Iterable<string>;
	constructor (base: string, args: Iterable<string>) {
		this.base = base;
		this.args = args;
	}
	generateNode(): string {
		return this.base + "<" + Merger.generateAll(this.args, new ValueMerger()) + ">"/*unknown*/;
	}
	isFunctional(): boolean {
		return false/*unknown*/;
	}
	isVar(): boolean {
		return false/*unknown*/;
	}
	generateBeforeName(): string {
		return ""/*unknown*/;
	}
	generateSimple(): string {
		return this.base/*unknown*/;
	}
	is(type: string): boolean {
		return "template".equals(type)/*unknown*/;
	}
}

File: src/java/magma/app/compile/type/TemplateNode.java
package magma.app.compile.type;

import magma.api.collect.list.Iterable;
import magma.app.compile.merge.Merger;
import magma.app.compile.merge.ValueMerger;
import magma.app.compile.node.Node;

public record TemplateNode(String base, Iterable<String> args) implements Node {
    public String generateNode() {
        return this.base + "<" + Merger.generateAll(this.args, new ValueMerger()) + ">";
    }

    public boolean isFunctional() {
        return false;
    }

    public boolean isVar() {
        return false;
    }

    public String generateBeforeName() {
        return "";
    }

    public String generateSimple() {
        return this.base;
    }

    public boolean is(String type) {
        return "template".equals(type);
    }
}
